quasar-core/src/main/java/co/paralleluniverse/fibers/instrument/CheckInstrumentationVisitor.java

```diff
@@ -69,6 +70,7 @@
     private ClassEntry classEntry;
     private boolean hasSuspendable;
     private boolean alreadyInstrumented;
+    private ArrayList<Pair<String,String>> methodsSyntheticStatic;
```

Comment:
Why not
```java
private final List<Pair<String, String> methodsSyntheticStatic;
```
You can then initialise it in the constructor and won't need to check whether it's been created every time. (This is for a `ClassVisitor`, so I don't see that always allocating one extra `ArrayList` per class is a hardship.)